core/server/proxy/src/main/java/alluxio/proxy/s3/S3ObjectTask.java

```diff
@@ -733,6 +739,7 @@ public Response continueTask() {
                   .setWriteType(S3RestUtils.getS3WriteType())
                   .putAllXattr(xattrMap).setXattrPropStrat(XAttrPropagationStrategy.LEAF_NODE)
                   .setOverwrite(true)
+                  .setCheckS3BucketPath(true)
```

Comment:
why is this check needed if we have the checkPathIsAlluxioDirectory step? it will anyway prevent any s3 create obj to create bucket folder in the inode tree is it?

Reply:
understood your point, actually in this case I have another question, in this case, would other calls gets a incorrect error response, say I delete the bucket from another proxy while this proxy hasn't expire the bucket path,  say like putobjecttagging, would it become a internal error  or NoSuchKey bcos of filenotexistexception from master instead of NoSuchBucket error?